Update coding style after "how we work" 2022-05-24 (#5762)
SonarSource/sonar-dotnet: docs/coding-style.md

  * especially when each branch ends with a `return` statement.
* Explicit `else` is not used after input validation.
* Each compiler directive outside method body (namely `#if`/`#endif`) should be preceded and followed by an empty line.

> * When there are multiple conditions to be met before the core logic of a method can be executed, we have a preference in structuring these conditions:

Reviewer: I would move this whole block one line higher - to have the if/else/logic related topics together. And the `#directives`  as a last point, as it is a different topic.

Reviewer: This sentence is quite wordy, I'd prefer to simplify the introduction to stay to the point
```suggestion
* For multiple conditions before the core method logic:
```
On the next paragraph, we'll need to continue with `chain` instead of `chained`.

>   * chained conditions in the same `if` statement together with positive logic ensure best readability (i.e. `if (condition) { doSomething(); }`) 

Reviewer: The sentence defines chained conditions, but the example doesn't illustrate that. Let's do something like this.
Also `DoSomething()` instead of `doSometing()` to follow coding style 😄 
```suggestion
  * chained conditions in the same `if` statement together with positive logic ensure best readability (i.e. `if (first && second) { DoSomething(); }`) 
```

  * when chained conditions cannot be used, we prefer early returns
  * otherwise, use nested conditions
* We prefer positive logic.

>   * Do use `if (result is {})` instead of `if (result != null)`. Consider also `if (result is not null)` (the `not` is more visible).

Reviewer: ```suggestion
  * Use `if (result is not null)` instead of `if (result != null)` and pattern matching if variable is needed `if (Something() is { } value)`.
```

Reviewer: Should this really be a sub-bullet point for the outer one?
We do prefer positive logic (I'd phrase it as positive conditions) with example `return !value.IsEmpty ? Something() : null`
And as a separate topic, we do prefer `is not null` and `is { }` as a null checks.

I'd probably promote it as a main bullet point.
